lib/bundler/cli/change.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Bundler
+  class CLI::Change
+    def initialize(options, gem_name)
+      @gem_name = gem_name
+      @options = options
+    end
+
+    def run
+      raise InvalidOption, "Please supply atleast one option to change." unless @options[:group] || @options[:version] || @options[:source]
+
+      definition = Bundler.definition
+
+      @dep = definition.dependencies.find {|d| d.name == @gem_name }
+
+      raise InvalidOption, "`#{@gem_name}` could not be found in the Gemfile." unless @dep
+
+      @pass_options = {}
+
+      initial_gemfile = IO.readlines(Bundler.default_gemfile)
+
+      set_group_options
+
+      set_version_options
+
+      @pass_options["source"] = @options[:source] if @options[:source]
```

Comment:
should this also handle path/git ?

Reply:
Yeah, it should. Adding them.

Reply:
Should they be editable?

Reply:
Also @segiddins , [these](https://github.com/bundler/bundler/blob/master/lib/bundler/dsl.rb#L18-L19) are valid keys we also need to persist them! I think we may have to change our approach?

Reply:
yeah we definitely need to persist all keys